src/pages/rechner/frage.tsx

```diff
@@ -19,47 +21,113 @@ function useLocalStorageState(key: string, defaultValue = ''): [string, React.Di
   }, [key, state]);
   return [state, setState];
 }
+const initialAnswerProp = '';
+const initialImpact = '';
 
-const IMPACT_AIRPLANE_HOUR_KG_CO2 = 0.29064;
-const MAX_QUESTION_NUMBER = 20;
+const Frage: NextPage<Props> = ({ questions }) => {
+  const [answer, setAnswer] = useLocalStorageState('answer', initialAnswerProp);
+  const [impact, setImpact] = useState(initialImpact);
+  const [questionIndex, setQuestionIndex] = useState(0);
+  const sortedQuestions = questions.sort((first, second) => first.category.localeCompare(second.category));
 
-const Frage: NextPage<Props> = ({ previousImpact = '', initialAnswer = '' }) => {
-  const [answer, setAnswer] = useLocalStorageState('answer', initialAnswer);
-  //ich glaub da müssti eigentlich epis anders im local storage denn mache => und de impact ganz am schluss hinzuefüege
-  const [impact, setImpact] = useLocalStorageState('impact', previousImpact);
+  const setLocalStorage = (key: string, value: string) => window.localStorage.setItem(key, value);
 
+  const getLocalStorage = (key: string) => window.localStorage.getItem(key);
+  const getImpactAndAnswerFromQuestionBefore = () => {
+    setQuestionIndex(questionIndex - 1),
+      setLocalStorage(questionIndex.toString(), answer),
+      getLocalStorage((questionIndex - 1).toString()) !== undefined
+        ? setAnswer(getLocalStorage((questionIndex - 1).toString()) || initialAnswerProp)
+        : '',
+      getLocalStorage((questionIndex - 1).toString()) !== undefined
+        ? setImpact(getLocalStorage('impact' + (questionIndex - 1).toString()) || initialImpact)
+        : '';
+  };
```

Comment:
I really don't understand the functions `getImpactAndAnswerFromQuestionBefore` and `saveImpactAndAnswer ` but I am pretty sure that they can't be right 😅 setting some state while simultaneously using that state to do stuff must result in pretty unexpected behaviour in combination with the react render lifecycles.

We should discuss this togheter.